include/circt/Dialect/FIRRTL/FIRRTLOps.h

```diff
@@ -24,27 +24,58 @@
 namespace circt {
 namespace firrtl {
 
+namespace direction {
+
+/// A type that indicates whether a port is an input or an output.
+enum Direction { Input, Output };
+
+/// The key in a module's attribute dictionary used to find the direction.
+const std::string attrKey = "portDirections";
+
+/// Perform an exclusive or with a Direction.  This has the effect of flipping
+/// the direction if \p b is true and leaving it unchanged if \p b is false.
+Direction add(Direction a, bool b);
```

Comment:
Why is this called "add"?  Why not `flip(a, doIt = true)` or something like that?

Reply:
Probably confusing. It's "adding" a flip (encoded as a boolean) to the current direction. I'll switch to your `direction::get` idea. There's two places that `add` is used in `LowerTypes` that I'll just sink there.

Reply:
Switched to just `direction::get` in rebase.